db/db.go

```diff
@@ -24,17 +28,59 @@ type Batch interface {
 	Write()
 }
 
+/*
+	Usage:
+
+	for itr.Seek(mykey); itr.Valid(); itr.Next() {
+		k, v := itr.Key(); itr.Value()
+		....
+	}
+*/
 type Iterator interface {
-	Next() bool
 
+	// Seek moves the iterator the position of the key given or, if the key
```

Comment:
So seek is like the replacement for IteratorPrefix?
Can you add in the example a check to show how to iterate in a range? Cuz if we use seek to find the beginning, there is probably an end we also want to see.

But yeah, Seek() with adding Prev() as well as  Next() seems like a nice generalization that allows a lot more flexibility to whoever uses this. Just show those example usages.

Reply:
No, the replacement for IteratorPrefix is db.IteratePrefix(db DB, prefix []byte) Iterator.

Reply:
Besides using db.IteratePrefix, You'll just have to manually stop iteration when you reach the end that you desire